src/analysis/deletion/removeZeroFluxRxns.m

```diff
@@ -0,0 +1,76 @@
+function modelOut = removeZeroFluxRxns(model, rxns)
+% Removes the reactions that cannot have any fluxes .
+% This function sets each reaction as the objective function and applies
+% FBA to find maximum and minuimum flux that can be carried through that
+% reaction, and removes the reactions that cannot have any fluxes.
+% notice: the results have functional dependency, not structural. results
+% may change due to a change in substrate for a model.
+%
+% USAGE:
+%    new_Model = removeZeroFluxRxns(model, reactions)
+%
+% INPUTS:
+%    model:                 COBRA model structure (must define `.rxns`, `.lb`, `.ub`)
+%
+%OPTIONAL INPUT
+%    reactions:             Cell array of reaction names in the model
+%
+% OUTPUT:
+%    new_model:             Curated model, the model with removed reactions
+%                           incapable of having a flux value, and corrected
+%                           upper and lower bounds.
+%
+% .. Author: - Farid Zare  5/31/2019
+
+if ~isstruct(model)
+    error('Input model must be structure');
+end
+
+if ~isfield(model, 'rxns')
+    error('Input model should contain `.rxns` field');
+end
+
+if ~isfield(model, 'lb')
+    error('Input model should contain `.lb` field');
+end
+
+if ~isfield(model, 'ub')
+    error('Input model should contain `.ub` field');
+end
+
+if nargin < 2
+    rxns = model.rxns;
+end
+
+modelOut = model;
+h = waitbar(0 , 'Working On Model... ');
+rxnsLength = length(rxns);
+
+for i = 1 : rxnsLength
+    
+    str = ['Working On Model...         ' + string(round(i / rxnsLength, 2) * 100) + '%'] ;
+    
+    if ishandle(h)
+        waitbar(i / rxnsLength, h, str);
+    else
+        warning('The process is terminated by the user');
+        break
+    end
+    
+    %Set each reaction as objective and apply FBA
+    model = changeObjective(model, rxns(i));
+    solMin = optimizeCbModel(model, 'min');
+    solMax = optimizeCbModel(model, 'max');
+    
+    %Set bounds to their feasible values in the network
+    rxnID = findRxnIDs(model, rxns(i) );
+    modelOut.lb(rxnID) = solMin.f;
+    modelOut.ub(rxnID) = solMax.f;
+    
+end
+
```

Comment:
You could simply run:
```[mins,maxs] = fluxVarability(model);
modelOut.lb = mins;
modelOut.ub = maxs;
```

Reply:
Dear  @tpfau  
Thank you so much for your valuable advice, I'm going through this comments and apply the changes !
I'm going to use [min, max] = fluxVariability(model , 0) to get the same result, and so there would be no for loop and I also have to terminate the waiting bar.
how about adding a waiting bar to fluxVariability itself? since it is usually time consuming, having an estimation of runtime would be helpful.

Reply:
There is a progress bar for some instances of fluxVariability. However its currently not implemented for parallel fluxVariability (which is the default mode, as this is not as trivial as it sounds since you need an external monitor. Maybe it will be implemented at a later stage.